### ISSUE TITLE
New pool implementation throws IllegalStateException with SlotState-Busy

### ISSUE THREAD

**Issue by misiekzap** [akka/akka-http#2254]
Akka HTTP throws IllegalStateException:
```
Connection was shutdown.
akka.http.impl.engine.client.pool.SlotState$BusyState$$anon$1: Connection was shutdown
```
when trying to connect to server which is down (or non-existing).
Mostly it returns "StreamTcpException" which is expected, when can't connect, but if more request happens (i.e. during retrial) - for example 100 or more akka throws IllegalStateException with BusyState and closes connection.
1. This issue happens only with HTTPS (for HTTP is working as expected)
2. This issue happens with new pool implementation (with legacy is working as expected).

Below code reproduces issue:

```
val config = ConfigFactory
        .parseString("""akka.http.host-connection-pool{
                      max-connections = 2
                      min-connections = 0
                      max-open-requests = 2
                      pool-implementation = new
                 }""".stripMargin)
        .withFallback(ConfigFactory.load())
      def sslContext: SSLContext = {
        val context = SSLContext.getInstance("TLS")
        context.init(Array(), Array(), new SecureRandom())
        context
      }
      val host = "non-existing-host"
      val port = 1
      val bufferRequests = 501
      val httpsContext: HttpsConnectionContext = new HttpsConnectionContext(sslContext)
      val poolSettings: ConnectionPoolSettings = ConnectionPoolSettings(config)
      println(poolSettings)

      val httpsReqRunner = Source
        .queue[(HttpRequest, Promise[HttpResponse])](bufferRequests, OverflowStrategy.dropNew)
        .via(Http().cachedHostConnectionPoolHttps(host, port, httpsContext, poolSettings))
//                  .via(Http().cachedHostConnectionPool(host, port, poolSettings))
        .toMat(Sink.foreach({
        case (Success(resp), p) => p.success(resp)
        case (Failure(ex), p) => p.failure(ex)
      }))(Keep.left).run

      val respFutures = for {
        n <- 1 to 200
      } yield {
        //        Thread.sleep(10)
        println(s"$n")

        val respPromise = Promise[HttpResponse]
        httpsReqRunner.offer(RequestBuilding.Get(Uri("/rest")) -> respPromise)
          .flatMap {
            case QueueOfferResult.Enqueued => respPromise.future
            case _                         => Future.failed(new Exception("BOOOM!!!"))
          }
          .recover {
            case x: StreamTcpException    => "cant connect" //println("cant connext: "+ x.getMessage)
            case x: BufferOverflowException => println("overflow:  " + x.getMessage); "boe"
//            case x : IllegalStateException => println("slot state: " + x) ; "ise"
            case ex: Throwable =>
              println(s"$n : Run request failed with unexpected exception: ${ex.getMessage}", ex)
              throw ex
          }
      }
      val responses = Future.sequence(respFutures).await
      println("responses: " + responses)
```

**Comment by jrudolph** [comment]
Hi @misiekzap,

which version was that? Did you check with 10.1.5?

**Comment by misiekzap** [comment]
Hi @jrudolph,

yes, I tried with 10.1.4 and 10.1.5 as well.

**Comment by misiekzap** [comment]
Issue seems to be resolved in Akka HTTP 10.1.8.
Thanks for good work!

**Comment by jrudolph** [comment]
Great, thanks for the update, @misiekzap.
